src/vm/eventpipeconfiguration.cpp

```diff
@@ -41,38 +38,32 @@ EventPipeConfiguration::~EventPipeConfiguration()
         EX_TRY
         {
             DeleteProvider(m_pConfigProvider);
-            m_pConfigProvider = NULL;
         }
         EX_CATCH {}
         EX_END_CATCH(SwallowAllExceptions);
     }
 
-    if (m_pProviderList != NULL)
+    // We swallow exceptions here because the HOST_BREAKABLE
+    // lock may throw and this destructor gets called in throw
+    // intolerant places. If that happens the provider list will leak
+    EX_TRY
     {
-        // We swallow exceptions here because the HOST_BREAKABLE
-        // lock may throw and this destructor gets called in throw
-        // intolerant places. If that happens the provider list will leak
-        EX_TRY
-        {
-            // Take the lock before manipulating the list.
-            CrstHolder _crst(EventPipe::GetLock());
-
-            SListElem<EventPipeProvider *> *pElem = m_pProviderList->GetHead();
-            while (pElem != NULL)
-            {
-                // We don't delete provider itself because it can be in-use
-                SListElem<EventPipeProvider *> *pCurElem = pElem;
-                pElem = m_pProviderList->GetNext(pElem);
-                delete (pCurElem);
-            }
+        // Take the lock before manipulating the list.
+        CrstHolder _crst(EventPipe::GetLock());
 
-            delete (m_pProviderList);
+        SListElem<EventPipeProvider *> *pElem = m_pProviderList->GetHead();
+        while (pElem != NULL)
```

Comment:
Perhaps modify this to use the `SLink` methods `IsEmpty` and `RemoveHead` to avoid manually walking the list.

Reply:
I would leave it for a future PR. I just removed an unnecessary check few lines below.